### PR TITLE
self instruct

### PR DESCRIPTION
**Pull request by StochasticRomanAgeev**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -50,7 +49,10 @@ dependencies = [
     "peft",
     "sentencepiece",
```
Comment (sarthaklangde) [inline pyproject.toml:50]: Remove one sentencepience

File: src/xturing/self_instruct/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Dataset generation
+
+Process:
```
Comment (sarthaklangde) [inline src/xturing/self_instruct/README.md:3]: @StochasticRomanAgeev should we delete this or improve this?

**Review by sarthaklangde — CHANGES_REQUESTED** [review]

File: src/turing/self_instruct/generate_instances.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+from turing.self_instruct.templates.instance_gen_template import (
+    input_first_template_for_gen,
+    output_first_template_for_clf,
+)
+
+random.seed(42)
+
+
+def generate_instances(
```
[the remark this answers is not in the document]
Comment (RomaA2000) [reply src/turing/self_instruct/generate_instances.py]: This file was accidentally added, new version operates with pathlib so readable errors

File: src/xturing/self_instruct/README.md (hunk 2 above)
Comment (RomaA2000) [reply src/xturing/self_instruct/README.md:3]: We can add reference to .ipynb, because we have the same text there

File: pyproject.toml (hunk 1 above)
Comment (RomaA2000) [reply pyproject.toml:50]: Done

File: examples/dataset_generation/create_alpaca_dataset.ipynb [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Generate dataset for ALPACA\n",
+    "\n",
+    "### Steps\n",
+    "\n",
+    "1. Bootstrap the instructions using\n",
```
Comment (sarthaklangde) [inline examples/dataset_generation/create_alpaca_dataset.ipynb:12]: Sentences are not correct. Can you fix this?

File: examples/dataset_generation/create_alpaca_dataset.ipynb [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "instruction_dataset = InstructionDataset.generate_dataset(api_key=\"<your api key>\", path=\"./seed_tasks.jsonl\")"
```
Comment (sarthaklangde) [inline examples/dataset_generation/create_alpaca_dataset.ipynb:122]: Rename this to

```suggestion
    "dataset = InstructionDataset.generate_dataset(api_key=\"<your api key>\", path=\"./seed_tasks.jsonl\")"
```

**Review by sarthaklangde — CHANGES_REQUESTED** [review]

File: examples/dataset_generation/create_alpaca_dataset.ipynb (hunk 5 above)
Comment (sarthaklangde) [inline examples/dataset_generation/create_alpaca_dataset.ipynb:122]: How does the user add organization? 

Also, how to use any other model except davinci?

Comment (sarthaklangde) [reply examples/dataset_generation/create_alpaca_dataset.ipynb:122]: Should we instead create an OpenAI models class from which you can choose? davinci, ada, etc.?

**Review by sarthaklangde — CHANGES_REQUESTED** [review]

File: examples/dataset_generation/create_alpaca_dataset.ipynb (hunk 5 above)
Comment (RomaA2000) [reply examples/dataset_generation/create_alpaca_dataset.ipynb]: engine argument is supported, so user can provide different model from OpenAI
also organization argument is supported, but for standard api key it is not needed
Added information how to set this parameters to .ipynb

Comment (RomaA2000) [reply examples/dataset_generation/create_alpaca_dataset.ipynb]: Done

**Pull request merged**